lib/matplotlib/colorbar.py

```diff
@@ -915,150 +904,34 @@ def _process_values(self, b=None):
-        """
-        if self.extend == 'neither':
-            y = np.linspace(0, 1, N)
-        else:
-            automin = automax = 1. / (N - 1.)
-            extendlength = self._get_extension_lengths(self.extendfrac,
-                                                       automin, automax,
-                                                       default=0.05)
-            if self.extend == 'both':
-                y = np.zeros(N + 2, 'd')
-                y[0] = 0. - extendlength[0]
-                y[-1] = 1. + extendlength[1]
-            elif self.extend == 'min':
-                y = np.zeros(N + 1, 'd')
-                y[0] = 0. - extendlength[0]
-            else:
-                y = np.zeros(N + 1, 'd')
-                y[-1] = 1. + extendlength[1]
-            y[self._inside] = np.linspace(0, 1, N)
-        return y
 
-    def _proportional_y(self):
-        """
-        Return colorbar data coordinates for the boundaries of
-        a proportional colorbar.
-        """
+        # otherwise values are set from the boundaries (and probably aren't
+        # that important)
         if isinstance(self.norm, colors.BoundaryNorm):
-            y = (self._boundaries - self._boundaries[0])
-            y = y / (self._boundaries[-1] - self._boundaries[0])
+            b = self.norm.boundaries
         else:
-            y = self.norm(self._boundaries.copy())
-            y = np.ma.filled(y, np.nan)
-        if self.extend == 'min':
-            # Exclude leftmost interval of y.
-            clen = y[-1] - y[1]
-            automin = (y[2] - y[1]) / clen
-            automax = (y[-1] - y[-2]) / clen
-        elif self.extend == 'max':
-            # Exclude rightmost interval in y.
-            clen = y[-2] - y[0]
-            automin = (y[1] - y[0]) / clen
-            automax = (y[-2] - y[-3]) / clen
-        elif self.extend == 'both':
-            # Exclude leftmost and rightmost intervals in y.
-            clen = y[-2] - y[1]
-            automin = (y[2] - y[1]) / clen
-            automax = (y[-2] - y[-3]) / clen
-        if self.extend in ('both', 'min', 'max'):
-            extendlength = self._get_extension_lengths(self.extendfrac,
-                                                       automin, automax,
-                                                       default=0.05)
-        if self.extend in ('both', 'min'):
-            y[0] = 0. - extendlength[0]
-        if self.extend in ('both', 'max'):
-            y[-1] = 1. + extendlength[1]
-        yi = y[self._inside]
-        norm = colors.Normalize(yi[0], yi[-1])
-        y[self._inside] = np.ma.filled(norm(yi), np.nan)
-        return y
+            # otherwise make the boundaries from the size of the cmap:
+            N = self.cmap.N + 1
+            b, _ = self._uniform_y(N)
+        # add extra boundaries if needed:
+        if self._extend_lower():
+            b = np.hstack((b[0] - 1, b))
+        if self._extend_lower():
```

Comment:
`_extend_upper()`

Reply:
Indeed, but per the parenthetical comment above, I don't _think_ boundaries are ever used except with `b[self.inner]`.  But I will double check...

Reply:
No they are used, but apparently not tested with anything other than `extend='both'`.

Reply:
Added test for `both`, `min`, `max`